cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/NodeBuilder.kt

```diff
@@ -269,3 +270,53 @@ fun <T : Node, S> T.codeAndLocationFrom(frontend: LanguageFrontend<S, *>, rawNod
 
     return this
 }
+
+fun <T : Node, S> T.codeAndLocationFromChildren(
+    frontend: LanguageFrontend<S, *>,
```

Comment:
It's a bit cumbersome to specify a frontend here, because we usually already have the frontend as part of the meta-data provider. We could use a context receiver for a `CodeAndLocationProvider` and we should change the `CodeAndLocationProvider` to implement `codeOf` and `locationOf` instead of the weird `setCodeAndLocation`. But that is probably a bigger change. If you want I can try to implement this here or I can try to do it after we merge this one?

Reply:
I would prefer to keep this PR small, as I need it for another. Changing it later sound interesting but for now I would update the Documentation of the function and then merge.